qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,346 @@
+            return False
+
+        if num_vars == 2 and rhs == 0:
+            if sense in (Constraint.Sense.LE, Constraint.Sense.GE):
+                # x-y<=0
+                # x-y>=0
+                return coeff_array.min() == -1.0 and coeff_array.max() == 1.0
+        elif num_vars >= 2:
+            if sense == Constraint.Sense.LE and rhs == 1:
+                if all(i == 1 for i in params.values()):
+                    # x1+x2+...<=1
+                    return True
+            elif sense == Constraint.Sense.GE and rhs == num_vars - 1:
+                if all(i == 1 for i in params.values()):
+                    # x1+x2+...>=n-1
+                    return True
+
+        return False
+
+    @staticmethod
+    def _auto_define_penalty(problem) -> float:
+        """Automatically define the penalty coefficient.
+
+        Returns:
+            Return the minimum valid penalty factor calculated
+            from the upper bound and the lower bound of the objective function.
+            If a constraint has a float coefficient,
+            return the default value for the penalty factor.
+        """
+
+        default_penalty = 1e5
+
+        # Check coefficients of constraints.
+        # If a constraint has a float coefficient, return the default value for the penalty factor.
+        terms = []
+        for constraint in problem.linear_constraints:
+            terms.append(constraint.rhs)
+            terms.extend(constraint.linear.to_array().tolist())
+        if any(isinstance(term, float) and not term.is_integer() for term in terms):
+            logger.warning(
+                "Warning: Using %f for the penalty coefficient because "
+                "a float coefficient exists in constraints. \n"
+                "The value could be too small. "
+                "If so, set the penalty coefficient manually.",
+                default_penalty,
+            )
+            return default_penalty
+
+        lin_b = problem.objective.linear.bounds
+        quad_b = problem.objective.quadratic.bounds
+        return 1.0 + (lin_b.upperbound - lin_b.lowerbound) + (quad_b.upperbound - quad_b.lowerbound)
+
+    def interpret(self, x: Union[np.ndarray, List[float]]) -> np.ndarray:
+        """Convert the result of the converted problem back to that of the original problem
+
+        Args:
+            x: The result of the converted problem or the given result in case of FAILURE.
+
+        Returns:
+            The result of the original problem.
+
+        Raises:
+            QiskitOptimizationError: if the number of variables in the result differs from
+                                     that of the original problem.
+        """
+
+        if len(x) != self._src_num_vars:
+            raise QiskitOptimizationError(
+                "The number of variables in the passed result differs from "
+                "that of the original problem."
```

Comment:
```suggestion
                f"The number of variables in the passed result ({len(x)}) differs from "
                f"that of the original problem ({self._src_num_vars})."
```
Might be nice to have the numbers in the error message to perhaps give some helpful extra info

Reply:
Thanks. It is helpful.